cmd/gqltool/main.go

```diff
@@ -40,10 +40,16 @@ func main() {
 }
 
 func getDefaultConfigHome() string {
-	// TODO(adamb): switch to os.UserConfigDir()
-	dir, err := os.UserHomeDir()
+	dir, err := os.UserConfigDir()
 	if err != nil {
 		panic(err)
 	}
-	return filepath.Join(dir, ".config", "stateful")
+	_, fErr := os.Stat(dir)
+	if os.IsNotExist(fErr) {
+		mkdErr := os.MkdirAll(dir, os.ModePerm)
```

Comment:
```suggestion
		mkdErr := os.MkdirAll(dir, 0o700)
```

The permissions you used ("ModePerm") is equivalent to `777` which is **extremely dangerous**. We should use `700` which allows all permissions but only for the user. (See [here](https://mason.gmu.edu/~montecin/UNIXpermiss.htm) for an explanation).